storage/s3.go

```diff
@@ -125,6 +135,20 @@ func (s *S3) Stat(ctx context.Context, url *url.URL) (*Object, error) {
 
 	etag := aws.StringValue(output.ETag)
 	mod := aws.TimeValue(output.LastModified)
+
+	if s.noSuchUploadRetryCount > 0 {
+		m := output.Metadata
+		if res, ok := m[META_DATA_S5CMD_RETRY_CODE_KEY]; ok {
```

Comment:
nitpick: I would prefer the name `metadata`for the variable or wouldn't define it as access to it is not that long.